google/gax/__init__.py

```diff
@@ -588,10 +590,10 @@ def result(self, timeout=None):
 
     def exception(self, timeout=None):
         """Similar to result(), except returns the exception if any."""
-        if self._poll(timeout).HasField('error'):
-            return self._operation.error
+        if not self._poll(timeout).HasField('error'):
```

Comment:
Actually, I noticed I was using the positive-if pattern in some places and the exceptional-if pattern in others inconsistently. I changed them all to the exceptional-if at first because that usually makes it easier to bring in new exceptional cases later. Changed it back because I wasn't sure if it was pythonic. Changing it back again with comments.